server/models/UserInstanceRole.ts

```diff
@@ -0,0 +1,24 @@
+import { Entity, JoinColumn, ManyToOne, PrimaryColumn } from 'typeorm';
+import { User } from './User';
+
+export type InstanceRoles = 'admin' | 'owner';
+
+@Entity({ name: 'user_instance_roles' })
+export class UserInstanceRole {
+  @PrimaryColumn()
+  user_id!: number;
+
+  @PrimaryColumn()
+  role_name!: InstanceRoles;
+
+  @ManyToOne(_type => User)
+  @JoinColumn({ name: 'user_id' })
+  user!: User;
+
+  constructor(params: { userId: number; roleName: InstanceRoles }) {
```

Comment:
This shouldn't be required if you have `if (params)`

Reply:
The `if (params)` is required for the migration; errors without them (apparently the constructor is called with no arguments?).  Ideally we keep it so that the params is not optional so people writing actual TS code don't think it's optional when it's not.